src/odemis/acq/millmng.py

```diff
@@ -372,12 +383,14 @@ def _milling_setting_iterator(self, site: CryoFeature):
 
                     # Estimate the drift
                     drift_est.estimate()
+                    self._drift = (pxs_anchor[0] * drift_est.drift[0],
+                                   -(pxs_anchor[1] * drift_est.drift[1]))
 
                     # Move FIB to compensate drift
                     previous_shift = self._scanner.shift.value
                     logging.debug("Ion-beam shift in m : %s", previous_shift)
-                    self._scanner.shift.value = (pxs_anchor[0] * drift_est.drift[0] + previous_shift[0],
-                                                 -(pxs_anchor[1] * drift_est.drift[1]) + previous_shift[1])  # shift in m - absolute position
+                    self._scanner.shift.value = (self._drift[0] + previous_shift[0],
+                                                 self._drift[1] + previous_shift[1])  # shift in m - absolute position
```

Comment:
There is something fishy here. `self._drift` is used  in `_restore_beam_shift()` to set the shift from the base shift (aka 0) to the current shift. However, here, scanner.shift should already contain the shift until the previous acquisition. So using drift_est.drift is fine here, but we should add up all the drifts since the beginning for the  `_restore_beam_shift()`.

Also, another difficulty is when the same current is used as the previous milling (task). In that case the current is not changed, so the shift says "good" (and the drift is only updated by a little bit). However, as soon as we change current, the shift is reset. => we need to reset the beam shift at the end of the task so that if another task is started after (without probe current change), the initial shift is the base shift.

Reply:
I have been thinking a little bit more about the issue that the beam shift at the beginning might not be the base shift. A possible simple way to deal with it is to have a "reset_beam_shift()" function that would be called at the very beginning and ensure that the beam shift is at the base value (by changing to a different probe current and changing back to the previous probe current).

Another alternative would be to hide all these difficulties in the orsay driver. The beam shift VA would be (only) the extra value we add on top of the base shift... Then all these tricks to handle shift reset would go away. We would still provide the base shifts control, but via a metadata. That feel a little late to do such a change... But maybe that's the best way to handle it. Let me know how you feel about it.